src/core/thread/mle.cpp

```diff
@@ -3282,9 +3282,14 @@ otError Mle::HandleParentResponse(const Message &aMessage, const Ip6::MessageInf
     }
 #endif
 
-    // if already have a candidate parent, only seek a better parent
-    if (mParentCandidate.GetState() == Neighbor::kStateParentResponse)
+    // Continue to process the "ParentResponse" if it is from current
+    // parent candidate to update the challenge and frame counters.
+
+    if ((mParentCandidate.GetState() == Neighbor::kStateParentResponse) &&
+        (mParentCandidate.GetRloc16() != sourceAddress.GetRloc16()))
```

Comment:
Should we match on extended address? RLOCs are only unique within a given partition.